src/include/common/object_pool.h

```diff
@@ -18,7 +19,7 @@ class AlignedByteAllocator {
    * @return a pointer to the byte array allocated.
    */
   T *New() {
-    auto *result = reinterpret_cast<T *>(new uint64_t[sizeof(T) / 8 + 1]);
+    auto *result = reinterpret_cast<T *>(new uint64_t[(sizeof(T) + 7) / 8]);
```

Comment:
Does it make sense to use posix_memalign here?

Reply:
problem with `posix_memalign` is that it's more of a C-style function. I think if the two are equivalent and not too much extra code is required we should stick to a more modern C++ code style.

I agree that this could stand to get some readability improvements though, either by using the Padding function in `StorageUtil` or comments.

Reply:
I understand what this is doing, but the code and intent would be much more transparent if we used posix_memalign. 

While this is technically C++, we are going to some effort to instruct the compiler and tell it what to do, which feels wrong.  IMO, we should use posix_memalign.